changes/issue3605.yaml

```
@@ -0,0 +1,2 @@
+fix:
+  - "Solve the collision of task_run_name and prefect context variables [#3605](https://github.com/PrefectHQ/prefect/issues/3605)"
```

Small tweak to the changelog entry. Also note we usually have a hyphen between the description and the PR link.

```suggestion
  - "Task arguments take precedence when generating `task_run_name` - [#3605](https://github.com/PrefectHQ/prefect/issues/3605)"
```